woocommerce/woocommerce-gutenberg-products-block
# Check block templates and parts directories to the Gutenberg 12.1.0 convention

With WordPress/gutenberg#36647, Gutenberg 12.1.0 has changed the convention for the directory paths from `block-templates` and `block-template-parts` to `templates` and `parts` respectively.

We are also moving to that convention but making sure that we remain backwards-compatible.

Fixes #5450

### Testing

#### Manual Testing

How to test the changes in this Pull Request:

> ⚠️ **Prerequisite:** you'll need to have [these changes](https://github.com/woocommerce/woocommerce/pull/31522) applied to WooCommerce Core for this PR to work as intended

1. Make sure your existing theme with the old convention (e.g. TT1 Blocks, using `block-templates`) still shows the templates correctly.
2. Change the directory names to `templates` and `parts` respectively, or download a theme which uses this new convention (e.g. [Tove](https://en-gb.wordpress.org/themes/tove/)), and make sure the templates still show correctly. 
3. Have both the old convention and the new convention living in the same theme and make sure that the templates under the newer names get prioritized instead of the deprecated ones.

### User Facing Testing
These are steps for user testing (where "user" is someone interacting with this change that is not editing any code).

### Changelog

> Update the template retrieving logic to allow for older Gutenberg convention and newer one (`block-templates`/`block-template-parts` vs. `templates`/`parts`)
